src/utils/cudnn.cpp

```diff
@@ -1315,7 +1332,7 @@ AlgoType find_best_heuristic_algorithm(
     if (p.memory > max_ws_size) {
       continue;
     }
-    algos.push_back(p.algo);
+    algos.push_back(cudnn_alg_config<AlgoType>(p.algo, p.mathType));
```

Comment:
Does this work?
```suggestion
    algos.emplace_back({p.algo, p.mathType});
```
Might be better?